### PR TITLE
[risk=low][RW-14070] Simplify egress email logic and add mail sending tool

### PR DESCRIPTION
**Pull request by jmthibault79**
To support the work of RW-14070:
* remove the separate "File name lengths" egress email - use the standard email in this case
* add a `./project.rb` tool to send emails, to help testing email changes

Tested locally with no observed changes.

I did not trigger the long file names egress event.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/exfiltration/EgressRemediationService.java [hunk 1]
```diff
@@ -50,27 +48,20 @@ public abstract class EgressRemediationService {
   private final LeonardoApiClient leonardoNotebooksClient;
   private final EgressEventAuditor egressEventAuditor;
   private final EgressEventDao egressEventDao;
-  private final UserAdminService userAdminService;
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/exfiltration/EgressRemediationService.java:53]: unused

File: api/src/main/java/org/pmiops/workbench/exfiltration/impl/EgressSumologicRemediationService.java [hunk 2]
```diff
@@ -76,12 +71,8 @@ public EgressSumologicRemediationService(
   @Override
   protected void sendEgressRemediationEmail(
       DbUser user, EgressRemediationAction action, DbEgressEvent event) throws MessagingException {
-    SumologicEgressEvent originalEvent = egressEventMapper.toSumoLogicEvent(event);
-    String environmentType =
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/exfiltration/impl/EgressSumologicRemediationService.java:80]: moved this into MailService

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 3]
```diff
@@ -380,23 +380,33 @@ public void sendBillingSetupEmail(DbUser dbUser, SendBillingSetupEmailRequest em
...
-            .build());
-  }
+      DbUser dbUser, EgressRemediationAction action, String serviceName) throws MessagingException {
+    String remediation = EGRESS_REMEDIATION_ACTION_MAP.get(action);
+    String environmentType =
+        appServiceNameToAppType(Strings.nullToEmpty(serviceName))
+            .map(LeonardoAppUtils::appDisplayName)
+            .orElse("Jupyter");
+
+    var substitutionMap =
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:390]: this is the former `getEgressRemediationEmailTemplate()`

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 4]
```diff
@@ -380,23 +380,33 @@ public void sendBillingSetupEmail(DbUser dbUser, SendBillingSetupEmailRequest em
...
+        Map.of(
+            EmailSubstitutionField.HEADER_IMG, getAllOfUsLogo(),
+            EmailSubstitutionField.ALL_OF_US, AOU_ITALICS,
+            EmailSubstitutionField.USERNAME, dbUser.getUsername(),
+            EmailSubstitutionField.EGRESS_REMEDIATION_DESCRIPTION, remediation,
+            EmailSubstitutionField.ENVIRONMENT_TYPE, environmentType);
+
+    String htmlMessage = buildHtml(EGRESS_REMEDIATION_RESOURCE, substitutionMap);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:398]: this is the former `sendEgressRemediationEmailWithContent()`

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 5]
```diff
@@ -588,24 +569,12 @@ private ImmutableMap<EmailSubstitutionField, String> initialCreditsExhaustionSub
         .build();
   }
 
-  private ImmutableMap<EmailSubstitutionField, String> initialCreditsExpirationSubstitutionMap(
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:591]: unused

File: api/tools/src/main/java/org/pmiops/workbench/tools/SendEmail.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+@Configuration
+@Import({
+  MailServiceImpl.class,
+  MandrillApi.class,
+})
+
+/* example usage:
```
Comment (jmthibault79) [inline api/tools/src/main/java/org/pmiops/workbench/tools/SendEmail.java:23]: new tool.  send yourself egress emails.

TODO: send other kinds of emails

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 7]
```diff
@@ -109,6 +109,7 @@ public class MailServiceImpl implements MailService {
 
   private static final String OPEN_LI_TAG = "<li>";
   private static final String CLOSE_LI_TAG = "</li>";
+  private static final String AOU_ITALICS = "<i>All of Us</i>";
```
Comment (evrii) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:112]: Good call

**Review by evrii — APPROVED** [review]
What a great tool to have! Thanks for putting this together. I generated both emails without issue.

**Pull request merged**